experimental/Parallel/src/Parallel.jl

```diff
@@ -0,0 +1,281 @@
+function getindex(wp::OscarWorkerPool, i::Int)
+  return wp.wids[i]
+end
+
+workers(wp::OscarWorkerPool) = wp.wids
+
+function get_channel(wp::OscarWorkerPool, id::Int; channel_size::Int=1024)
+  chnl = get!(wp.oscar_channels, id) do
+    RemoteChannel(()->Channel{Any}(channel_size), id)
+  end
+end
+
+# extra functionality
+
+function close!(wp::OscarWorkerPool)
+  for id in take!(wp)
+    rmprocs(id)
+  end
+  wp.wids = Int[]
+end
+
+# extend functionality so that `pmap` works with Oscar stuff
+
+function put_type_params(wp::OscarWorkerPool, a::Any)
+  for id in wp.wids
+    put_type_params(get_channel(wp, id), a)
+  end
+end
+
+#=
+# Custom implementation of `pmap`. This is potentially deprecated since we can also overwrite `remotecall_fetch` below. 
+function pmap(f::Any, wp::OscarWorkerPool, args0::Vector; kill_workers::Bool=false, wait_period=0.1)
+  !isready(wp) && is_zero(nworkers(wp)) && error("can not do computations on empty pool")
+  args = copy(args0)
+  n = nworkers(wp)
+  futures = Dict{Int, Pair{Int, Future}}()
+  results = Dict{Int, Any}() # Results might come back in different order
+  i = length(args0) + 1
+  while !isempty(args) || !isempty(futures)
+    # distribute some new work
+    while !isempty(args) && isready(wp)
+      id = take!(wp)
+      arg = pop!(args)
+      i = i - 1
+      put_type_params(get_channel(wp, id), arg) # send the parents up front
+      futures[id] = i => remotecall(f, id, arg)
+    end
+    
+    # gather results
+    for (id, (i, fut)) in futures
+      isready(fut) || continue
+      results[i] = fetch(fut)
+      delete!(futures, id)
+      if kill_workers && isempty(args)
+        rmprocs(id)
+      else
+        put!(wp, id)
+      end
+    end
+    sleep(wait_period)
+  end
+
+  @assert is_one(i)
+  # assemble the final result
+  return [results[i] for i in 1:length(args0)]
+end
+=#
+
+@doc raw"""
+    compute_distributed!(ctx, wp::OscarWorkerPool; wait_period=0.1)
+
+Given an `OscarWorkerPool` `wp` and a context object `ctx`, distribute tasks 
+coming out of `ctx` to be carried out on the workers (based on availability) and 
+process the result.
+
+For this to work, the following methods must be implemented for `ctx`:
+  - `pop_task!(ctx)` to either return a `Tuple` `(task_id::Int, func::Any, args)`, in which case `remotecall(func, wp, args)` will be called to deploy the task to the workers, or return `nothing` to indicate that all tasks in `ctx` have been exhausted, or return an instance of `WaitForResults` to indicate that some information on results of tasks which have already been given out is needed to proceed with giving out new tasks.  
+  - `process_result!(ctx, task_id::Int, res)` to process the result `res` of the computation of the task with id `task_id`. 
+
+Note: The programmer themself is responsible to deliver `task_id`s which are unique for the respective tasks!
+
+The return value is the `ctx` in its current state after computation.
```

Comment:
I am in doubt about the meaning of this line. Please clarify whether the computation runs until ctx is exhausted or whether it runs a single task popped from ctx and then returns or maybe something else(?).

(Yes I can guess from the code that you exhaust ctx and then return `nothing` as the value of ctx, but even a user who ponders about experimenting with this, might not pass beyond the docstrings, if he feels discouraged here. )